### ISSUE TITLE
Squiz.Commenting.FunctionComment truncates class name if it contains underscores

### ISSUE THREAD

**Issue by mahmoudsaeed** [PHPCSStandards/PHP_CodeSniffer#1197]
## Describe the bug

The `Squiz.Commenting.FunctionComment` sniff truncates class name if it contains underscores.

### Code sample
```php
/**
 * @return integer|WP_Error
 */
function test()
{
    return 0;
}
```

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="CS">
  <rule ref="Squiz.Commenting.FunctionComment"/>
</ruleset>
```

### To reproduce
1. Create a file called `test.php` with the code sample above
2. Run `phpcbf test.php`
3. Check out the modified code. It corrected the `integer` type to `int`, but it truncated the class name `WP_Error` to just `WP`:
```php
/**
 * @return int|WP
 */
function test()
{
    return 0;
}
```

## Expected behavior

The code after modification should look like:
```php
/**
 * @return int|WP_Error
 */
function test()
{
    return 0;
}
```

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System | macOS 15.6 |
| PHP version  | 8.4.7  |
| PHP_CodeSniffer version | 3.13.0 |
| Standard | Squiz |
| Install type | Composer |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
Thanks for testing @mahmoudsaeed !
